traitsui/qt4/check_list_editor.py

```diff
@@ -132,11 +151,41 @@ def update_editor(self):
             pass
 
 
-class CustomEditor(SimpleEditor):
+class CustomEditor(BaseCheckListEditor):
     """ Custom style of editor for checklists, which displays a set of check
         boxes.
     """
 
+    #: List of tuple(signal, slot) to be disconnected in dispose
+    _connections_to_dispose = List(Tuple(Any, Callable))
+
+    #: List of tuple(signal, slot) to be disconnected while rebuilding the
+    #: editor.
+    _connections_to_rebuild = List(Tuple(Any, Callable))
+
+    def init(self, parent):
+        """ Finishes initializing the editor by creating the underlying toolkit
+            widget.
+        """
+        self.create_control(parent)
+        super(CustomEditor, self).init(parent)
+
+    def dispose(self):
+        """ Disposes of the contents of an editor.
+        """
+        while self._connections_to_rebuild:
+            signal, handler = self._connections_to_rebuild.pop()
+            signal.disconnect(handler)
+
+        while self._connections_to_dispose:
+            signal, handler = self._connections_to_dispose.pop()
+            signal.disconnect(handler)
+
+        if self._mapper is not None:
+            self._mapper = None
+
+        super().dispose()
```

Comment:
In the other PRs, I decided to just call `Editor.dispose(self)` and leave #884 for another day. I will do that here too to be consistent.